### ISSUE TITLE
Changing confirm prompt so Ctrl+C exits the process   

### ISSUE THREAD

**Issue by tiagonapoli** [terkelg/prompts#252]
### Is your feature request related to a problem?
In my use case when pressing <kbd>CTRL</kbd>+<kbd>C</kbd> on a prompt the program should immediately terminate. Using the default confirm prompt I noted that it would answer `yes` and continue the program, so I implemented the following solution - is there a native one?

```typescript
import prompts from 'prompts'

interface PromptState {
  aborted: boolean
}

const enableTerminalCursor = () => {
  process.stdout.write('\x1B[?25h')
}

const onState = (state: PromptState) => {
  if (state.aborted) {
    // If we don't re-enable the terminal cursor before exiting
    // the program, the cursor will remain hidden
    enableTerminalCursor()
    process.stdout.write('\n')
    process.exit(1)
  }
}

export const promptConfirm = async (message: string, initial = true): Promise<boolean> => {
  const { response } = await prompts([{ message, initial, onState, type: 'confirm', name: 'response' }])
  return response
}
```

**Comment by terkelg** [comment]
Thank you @tiagonapoli. Currently there are no native solutions to this. It should be an option in the next major version of prompts 👍

**Comment by lunaboona** [comment]
> 
> 
> Thank you @tiagonapoli. Currently there are no native solutions to this. It should be an option in the next major version of prompts 👍

This is great to hear since I'm currently facing the same issue! Woudl you happen to have aroadmap or any estimates for the next version?

**Comment by terkelg** [comment]
sorry, I don't have any timeline at the moment. I'm quite busy working on other projects 👍

**Comment by elliot-nelson** [comment]
An alternate suggestion: if you have a lot of prompts, it might be better to make an onState that throws an error that can bubble up to the top, rather than immediately hard-exiting the program.

```js
class AbortedError extends Error { }

function onState(state) {
    if (state.aborted) throw new AbortedError();
}
```

Now you have a reusable onState you can attach to all of your prompts, and in the outermost catch of your application, you can choose to:

```
main().catch(error => {
    if (!(error instanceof AbortedError)) {
        console.error(error.stack);
    }
    process.exitCode = 1;
});
```

This way the `prompts` package will clean up terminal stuff itself, because you aren't exiting the process before it can clean up, and your CLI program exits with an exit code 1 a little more gracefully (in my opinion).

**Comment by elliot-nelson** [comment]
Update: I should have tested my suggestion first, it looks like throwing from inside onState() actually ends up aborting the process anyway because you are inside a readline callback.  So my "graceful" claim is not true today. 🥩

**Comment by Blazzike** [comment]
My recommended solution:

```js
onState: (state) => {
  if (state.aborted) {
    process.nextTick(() => {
      process.exit(0);
    })
  }
}
```

**Comment by ChocolateLoverRaj** [comment]
@tiagonapoli Maybe there could be an API similar to this:
```js
import prompts from 'prompts'

prompts.setExitOnAbort(true)
```

Or this:
```js
import prompts from 'prompts'

prompts.overrideAborting(false)
```

> sorry, I don't have any timeline at the moment. I'm quite busy working on other projects 👍

@terkelg Can we (me or @tiagonapoli or anyone) implement these changes and make a pull request?

**Comment by raineorshine** [comment]
Please don't add a procedural interface that mutates the module instance. It's more error-prone and messy if you want to need to revert the behavior back.

If anything you could add `exitOnAbort: true` as a prompt option, but I've come to prefer @Blazzike's solution since it gives you control over the exit code. Or you could do something like `exitOnAbort: 0`, `exitOnAbort: 1`, etc. (Personally I think `exitOnAbort: 1` should be the default.)

**Comment by TehBrian** [comment]
> If anything you could add `exitOnAbort: true` as a prompt option,

Hard-coding the only two options as either exit or don't exit would be very limiting. Even something as simple as being able to log a short message to console upon Ctrl+C would provide much more flexibility. Preferably, you'd just be able to pass in your own function.

Although I like @Blazzike's solution for how intuitive and easy to implement it seems, I feel as if it would get excessively verbose if you had to repeat the same code every single prompt object.

Just throwing some ideas out there, perhaps we could allow you to set a global `onState` function that is called every prompt object as well?
```ts
import prompts from "prompts";

prompts.onState = (state) => {
    if (state.aborted) {
        process.nextTick(() => {
            console.log("See ya!");
            process.exit(0);
        })
    }
}
```

This would be mutating the module instance, though. Perhaps a nice middle-ground between configuring the module instance and per-prompt would be passing an option to `prompts()`?



> prompts.overrideAborting(false)

This is a neat idea! I'd imagine this would just

**Comment by TehBrian** [comment]
> If anything you could add `exitOnAbort: true` as a prompt option,

Hard-coding the only two options as either exit or don't exit would be very limiting. Even something as simple as being able to log a short message to console upon Ctrl+C would provide much more flexibility. Preferably, you'd just be able to pass in your own function.

Although I like @Blazzike's solution for how intuitive and easy to implement it seems, I feel as if it would get excessively verbose if you had to repeat the same code for every single prompt object.

Just throwing some ideas out there, perhaps we could allow you to set a global `onState` function that's called every prompt object?
```ts
import prompts from "prompts";

prompts.onState = (state) => {
    if (state.aborted) {
        process.nextTick(() => {
            console.log("See ya!");
            process.exit(0);
        })
    }
}
```

This would be mutating the module instance, though. Perhaps a nice middle-ground between configuring the module instance and configuring the `onState` per-prompt would be passing an option to `prompts()`?

```ts
var response = await prompts(questions, { onState: (state) => /* do something */ });

// or perhaps..

var response = await prompts(questions, { exitOnAbort: true });
```

**Comment by MMK21Hub** [comment]
I have this same problem. Unfortunately, pressing `esc` is also treated as a abortion, but I only want Ctrl+C to exit the whole thing. I don't believe that there is any provided way to do this.

**Comment by andersaamodt** [comment]
Yes, is there a workaround? Maybe it is possible to remove Prompt's SIGINT (Ctrl-C?) event listener, or to attach a new event listener that catches SIGINT preferentially? I tried to do this and failed.

**Comment by jedwards1211** [comment]
@terkelg why does prompts intercept any signals in the first place?  It seems like a very strange design choice to me, but maybe there's something I'm not thinking about...

**Comment by calebeby** [comment]
Prompts does not directly intercept signals. It does set the input to raw mode (to listen for individual keypresses). In raw mode the default ctrl+c handling doesn't work.

Here is the workaround I am using:

```js
// Handle Ctrl+C even when raw mode is used (for prompts)
process.stdin.on("keypress", function (_chunk, key) {
  if (key && key.name === "c" && key.ctrl) {
    process.exit(130);
  }
});
```

Based on this stack overflow answer: https://stackoverflow.com/questions/20165605/detecting-ctrlc-in-node-js

**Comment by jedwards1211** [comment]
@calebeby  oh, thanks, I didn't think about that. It would be good for prompts to at least provide an option to handle Ctrl-C by doing `process.kill(process.pid, 'SIGINT')`

**Comment by VibingCreator** [comment]
Works so far.

```ts
process.stdin.on("data", (data) => {
  const readableBuffer = data.toString();

  // CTRL + C
  if (readableBuffer === "\u0003") {
    process.exit();
  }
});
```

**Comment by bybraveHQ** [comment]
This is now the default in [@bybrave/prompts2](https://github.com/bybraveHQ/prompts2), a maintained drop-in fork: Ctrl+C restores the terminal (cursor shown, raw mode off) and then re-raises SIGINT, so the process terminates — or your own SIGINT handler runs. Per-question opt-out with `exitOnCtrlC: false` keeps the old soft-cancel behaviour. Esc/Ctrl+D remain soft cancels. `npm i @bybrave/prompts2`
